server-client/index.js

```diff
@@ -201,34 +201,18 @@ export class ServerClient {
 
     let type = action.type
     if (type === 'logux/subscribe' || type === 'logux/unsubscribe') {
-      return true
+      this.app.onAction(action, meta, this.clientId)
+      return false
     }
 
     let processor = this.app.getProcessor(type)
     if (!processor) {
-      this.app.internalUnkownType(action, meta)
+      this.app.internalUnkownType(action, meta, this.clientId)
       return false
     }
 
-    try {
-      let result = await processor.access(ctx, action, meta)
-      if (this.app.unknownTypes[meta.id]) {
-        delete this.app.unknownTypes[meta.id]
-        this.app.finally(processor, ctx, action, meta)
-        return false
-      } else if (!result) {
-        this.app.finally(processor, ctx, action, meta)
-        this.denyBack(action, meta)
-        return false
-      } else {
-        return true
-      }
-    } catch (e) {
-      this.app.undo(action, meta, 'error')
-      this.app.emitter.emit('error', e, action, meta)
-      this.app.finally(processor, ctx, action, meta)
-      return false
-    }
+    this.app.onAction(action, meta, this.clientId)
+    return false
```

Comment:
Not really like that, we return always `false` and re-add actions later. It looks like too hacky.

Can `this.app.onAction()` return `Promise<boolean>`?

Reply:
Should we return `this.app.onAction()` from filter?
We can change `onAction` to return a Promise, but we will have to resolve it to `false` unless we also change the `core`, because `syncMessage` would add duplicate action to log (https://github.com/logux/core/blob/main/sync/index.js#L89). Would it be ok to resolve it to `false`?

Reply:
> because syncMessage would add duplicate action to log

I mean remove `this.app.log.add()` from queue’s code and return `Promise<true|false>` instead from queue.

Reply:
I `onAction` to return promise, but currently it may be problematic for `sendSynced in `core`, maybe some changes can be made in `core/sync` to synchronize actions faster witch queues.

Reply:
What problem of `sendSynced` do you mean? It should be sent to client when the server will add it to the own log (and server need to call `access` first). So we need to wait the queue first.

Reply:
This is not a huge concern, but I meant that `sendSynced` for action can take a lot of time if client sends several actions at once.